### DISCUSSION TITLE
Support for Rich?

### DISCUSSION THREAD

**Discussion by harkabeeparolus** [tqdm/tqdm#1375] [Ideas]
Hi,

I've used tqdm for a few of my own toy projects previously. These days, [Rich](https://github.com/Textualize/rich#readme) is all the rage at least in certain parts of the Python community, and it certainly does have lots of colours, pretty progress bars, spinners, and live displays. However, tqdm has a good API, smoothing, good ETA calculations, and a great command line tool.

Basically in my humble opinion, Rich looks prettier and is more modular for building terminal stuff -- but tqdm shows me better data, and has a great CLI wrapper. I can't help but wonder if these projects couldn't help each other out somehow. 😊

- What are the chances of making a [Rich backend](https://rich.readthedocs.io/en/latest/progress.html) to display progress bars in a terminal?

- Or what about improving compatibility with Rich some other way, off the top of my head perhaps being able to display tqdm as a Rich renderable component inside a [live display](https://rich.readthedocs.io/en/latest/live.html)?

I haven't really fully thought this through, and I haven't put in the work to read and understand the tqdm source. But I just wanted to put this idea out there to see what people think. 😊

Cheers!

**Comment by casperdcl** [comment]
already done in [`tqdm.rich`](https://tqdm.github.io/docs/rich/)?

**Reply by harkabeeparolus** [reply]
Wow, so it is! I didn't see that. 😊 Thanks! I'll try it out.

**Reply by cnguyen10** [reply]
Is there a way to do nested `tqdm.rich` progress bar? The current implementation seems to allow only one `tqdm.rich` progress bar.
